app/components/publish-actions.tsx

```diff
@@ -149,9 +160,23 @@ const Embed = ({ chartWrapperRef, configKey, locale }: PublishActionProps) => {
                   "For embedding visualizations in systems without JavaScript support (e.g. WordPress).",
               })}
             />
+            <EmbedToggleSwitch
+              value="without-border"
+              checked={withoutBorder}
+              onChange={(_, checked) => handleStylingChange(checked)}
```

Comment:
At the top, `handleStylingChange` is wrapped in `useEventCallback`, but here we have a function that's re-created on each render (`(_, checked) => ...`). In order to improve the caching, I'd change the `handleStylingChange` callback to accept an event as the first function argument, and only pass `onChange={handleStylingChange}` here (see how it's used in the line 152 in `handleChange`, where we don't create another arrow function) 👍